docsy(v2): Ray images must ship wget; document head node resource floor (#1458)
unionai/docs: content/integrations/ray/_index.md

```python
image = (
    flyte.Image.from_debian_base(name="ray")
    .with_apt_packages("wget")
    .with_pip_packages("ray[default]==2.46.0", "flyteplugins-ray")
)
```

> [!WARNING] Your image must include `wget`
> KubeRay's readiness and liveness probes for the head pod shell out to `wget`
> to poll the raylet and GCS health endpoints. If the image has no `wget`, both
> probes fail permanently with `wget: command not found`, the head pod never
> reports `Ready`, the workers stay parked in their `wait-gcs-ready` init
> container, and the job is never submitted. The run sits in `Queued` with no

> > error message. Install it with `.with_apt_packages("wget")`, or use a base

Reviewer: Confirmed, and worth taking: the probe runs under `bash -c`, so the kubelet event carries the shell prefix. A reader pasting the event text into search should land on this callout.

```suggestion
> probes fail permanently with `bash: line 1: wget: command not found`, the head pod never
```

| `limits` | `Resources` | Resource limits for the head node |
| `pod_template` | `PodTemplate` | Full pod template (mutually exclusive with `requests`/`limits`) |

The head node runs the Ray dashboard, which starts nine subprocess modules on
top of GCS and the raylet. Give it at least 2 CPU and 4Gi of memory:

```python
head_node_config=HeadNodeConfig(
    requests=flyte.Resources(cpu=2, memory="4Gi"),
)
```

With less (1 CPU / 1000Mi, for example), `ray start --head` exits 1 and KubeRay

> recycles the head pod in a loop, so the cluster never becomes ready.

Reviewer: nit: the 2 CPU / 4Gi floor and the `exits 1` threshold are the one part of this page I could not ground. The nine-module dashboard fact is correct at `ray[default]==2.46.0`, but it does not fix a number; and `1000Mi` is one tenant's default (EKS), not the platform's, so the contrast does not generalize. Ray's real failure also raises the tail of `dashboard.log`, not a fixed message.

Suggest grounding it in the failure you actually observed rather than stating a threshold:

```suggestion
Under-provisioning the head node is a common cause of a cluster that never becomes
ready: if the dashboard cannot start, `ray start --head` fails and KubeRay recycles
the head pod in a loop. A head pod at 1 CPU and 1000Mi has been observed failing
this way.
```

Same applies to "at least" on L107.

| `limits` | `Resources` | Resource limits for the head node |
| `pod_template` | `PodTemplate` | Full pod template (mutually exclusive with `requests`/`limits`) |

The head node runs the Ray dashboard, which starts nine subprocess modules on
top of GCS and the raylet. Give it at least 2 CPU and 4Gi of memory:

```python
head_node_config=HeadNodeConfig(
    requests=flyte.Resources(cpu=2, memory="4Gi"),

> )

Reviewer: The premise here is not right: the snippet does parse. `head_node_config=HeadNodeConfig(...)` at module level is an assignment statement, not a bare keyword argument (`ast.parse` returns an `Assign`).

The real issue is narrower: it *reads* as a keyword argument, and it binds a name nothing uses. Showing it inside the call fixes both and matches the example above at L53-70. Note `worker_node_config` is required, so a minimal wrapper needs it.

````suggestion
```python
ray_config = RayJobConfig(
    head_node_config=HeadNodeConfig(requests=flyte.Resources(cpu=2, memory="4Gi")),
    worker_node_config=[WorkerNodeConfig(group_name="ray-group", replicas=2)],
)
```
````
